CoreOpenTibiaServer/src/COTS.GameServer/World/Loading/WorldLoader.TileParsing.cs

```diff
@@ -37,164 +37,103 @@ public static partial class WorldLoader {
+                throw new MalformedTileAreaNodeException("Unknow tile area node type.");
 
-        private static void ParseHouseTile(ParsingTree parsingTree, ParsingNode tileNode, ushort areaStartingX, ushort areaStartingY, byte areaZ) {
-            var stream = new WorldParsingStream(parsingTree, tileNode);
+            var nodeStream = new WorldParsingStream(parsingTree, tileNode);
 
-            // Calculating the tile's absolute coordinates
-            var xOffset = stream.ReadByte();
-            var yOffset = stream.ReadByte();
-            var tileX = (UInt16)(areaStartingX + xOffset);
-            var tileY = (UInt16)(areaStartingY + yOffset);
+            var tileX = areaStartingX + nodeStream.ReadByte();
+            var tileY = areaStartingY + nodeStream.ReadByte();
 
-            // Getting reference to house 
+            Tile tile = null;
+            if(tileNode.Type == NodeType.HouseTile)
+                tile = ParseHouseTile(ref nodeStream, (ushort)tileX, (ushort)tileY, areaZ); // Improve this, remove casts
+
+            var tileFlags = ParseTileAttributes(ref parsingTree,ref nodeStream, ref tile, tileNode);
+            if (tile != null)
+                tile.Flags.AddFlags(tileFlags);
+        }
+        
+        // Maybe it should be bool. Passing house by reference
+        private static Tile ParseHouseTile(ref WorldParsingStream stream, UInt16 tileX, UInt16 tileY, byte tileZ) {
             var houseId = stream.ReadUInt32();
-            var house = HouseManager.Instance.CreateHouseOrGetReference(houseId);
-            var tile = new HouseTile(
-                    x: tileX,
-                    y: tileY,
-                    z: areaZ,
-                    house: house);
+            House house = HouseManager.Instance.CreateHouseOrGetReference(houseId);
+
+            var tile = new HouseTile(tileX, tileY, tileZ, house);
             house.AddTile(tile);
 
+            return tile;
+        }
 
-            TileFlags updatedTileFlags = TileFlags.None;
+        private static TileFlags ParseTileAttributes(ref ParsingTree parsingTree, ref WorldParsingStream stream, ref Tile tile, ParsingNode tileNode) {
+            TileFlags tileFlags = new TileFlags();
+            NodeAttribute nodeAttribute;
+            while(!stream.IsOver) {
+                nodeAttribute = (NodeAttribute) stream.ReadByte();
+                switch(nodeAttribute) {
+                    case NodeAttribute.TileFlags:
+                        tileFlags = ParseTileFlags(ref stream);
+                        break;
+                    case NodeAttribute.Item:
+                        ParseTileItem(ref stream, ref tile, tileNode.Type == NodeType.HouseTile);
+                        break;
+                    default:
+                        throw new MalformedTileNodeException("Unknow node attribute " + nameof(nodeAttribute) + " of type " + nodeAttribute);
+                }
```

Comment:
Not returning tileFlags. And tileFlags should init as TileFlags.None. I will fix these in next commits